scripts-available/CDB_Groups.sql

```diff
@@ -0,0 +1,138 @@
+    RETURNS VOID AS $$
+DECLARE
+    cdb_group_role TEXT;
+    cdb_user_role TEXT;
+BEGIN
+    cdb_group_role := cartodb._CDB_Group_GroupRole(group_name);
+    cdb_user_role := cartodb._CDB_User_RoleFromUsername(username);
+    EXECUTE 'GRANT "' || cdb_group_role || '" TO "' || cdb_user_role || '"';
+END
+$$ LANGUAGE PLPGSQL;
+
+-- Removes a user from a group
+CREATE OR REPLACE
+FUNCTION cartodb.CDB_Group_RemoveMember(group_name text, username text)
+    RETURNS VOID AS $$
+DECLARE
+    cdb_group_role TEXT;
+    cdb_user_role TEXT;
+BEGIN
+    cdb_group_role := cartodb._CDB_Group_GroupRole(group_name);
+    cdb_user_role := cartodb._CDB_User_RoleFromUsername(username);
+    EXECUTE 'REVOKE "' || cdb_group_role || '" FROM "' || cdb_user_role || '"';
+END
+$$ LANGUAGE PLPGSQL;
+
+-- Grants table read permission to a group
+CREATE OR REPLACE
+FUNCTION cartodb.CDB_Group_Table_GrantRead(group_name text, username text, table_name text)
+    RETURNS VOID AS $$
+DECLARE
+    cdb_group_role TEXT;
+BEGIN
+    cdb_group_role := cartodb._CDB_Group_GroupRole(group_name);
+    EXECUTE 'GRANT USAGE ON SCHEMA "' || username || '" TO "' || cdb_group_role || '"';
+    EXECUTE 'GRANT SELECT ON TABLE "' || username || '"."' || table_name || '" TO "' || cdb_group_role || '"';
+END
+$$ LANGUAGE PLPGSQL;
+
+-- Grants table write permission to a group
+CREATE OR REPLACE
+FUNCTION cartodb.CDB_Group_Table_GrantReadWrite(group_name text, username text, table_name text)
+    RETURNS VOID AS $$
+DECLARE
+    cdb_group_role TEXT;
+BEGIN
+    cdb_group_role := cartodb._CDB_Group_GroupRole(group_name);
+    EXECUTE 'GRANT USAGE ON SCHEMA "' || username || '" TO "' || cdb_group_role || '"';
+    EXECUTE 'GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE "' || username || '"."' || table_name || '" TO "' || cdb_group_role || '"';
+END
+$$ LANGUAGE PLPGSQL;
+
+-- Revokes all permissions on a table from a group
+CREATE OR REPLACE
+FUNCTION cartodb.CDB_Group_Table_RevokeAll(group_name text, username text, table_name text)
+    RETURNS VOID AS $$
+DECLARE
+    cdb_group_role TEXT;
+BEGIN
+    cdb_group_role := cartodb._CDB_Group_GroupRole(group_name);
+    EXECUTE 'REVOKE ALL ON TABLE "' || username || '"."' || table_name || '" FROM "' || cdb_group_role || '"';
+END
+$$ LANGUAGE PLPGSQL;
+
+-----------------------
+-- Private functions
+-----------------------
+-- Given a group name returns a role. group_name must be a valid PostgreSQL idenfifier. See http://www.postgresql.org/docs/9.2/static/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
+CREATE OR REPLACE
+FUNCTION cartodb._CDB_Group_GroupRole(group_name text)
+    RETURNS TEXT AS $$
+DECLARE
+    group_role TEXT;
+BEGIN
+    IF group_name !~ '^[a-zA-Z_][a-zA-Z0-9_]*$'
+    THEN
+        RAISE EXCEPTION 'Group name (%) must be a valid identifier. See http://www.postgresql.org/docs/9.2/static/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS', group_name;
+    END IF;
+    RETURN 'g_' || md5(current_database()) || '_' || group_name;
```

Comment:
Ok, without `md5` then.

I've added a check at `cartodb._CDB_Group_GroupRole` with a user-friendly error message: "Group name should be shorter. Resulting role must have less than 60 characters, but it is longer: test_organizations_g_a123456789012345678901234567890123456789012345678901234567890". This way we don't have to propagate error handling and all generated roles will be valid.